File: python/raindrops/raindrops.py
def convert(number):
    is_factor_dict = {"Pling": number % 3 == 0, "Plang": number % 5 == 0, "Plong": number % 7 == 0}
    raindrops = "".join([k for k, v in is_factor_dict.items() if v])
    if raindrops != "":
        return raindrops
    else:
        return str(number)

File: r/raindrops/raindrops.R
raindrops <- function(number) {
  sounds <- list("Pling", "Plang", "Plong")
  mask <- c(number %% 3 == 0, number %% 5 == 0, number %% 7 == 0)
  ifelse(any(mask), do.call(paste0, sounds[mask]), as.character(number))
}
